rust/lance-linalg/src/simd/f32.rs

```diff
@@ -831,7 +831,7 @@ mod tests {
     #[test]
     fn test_f32x16_cmp_ops() {
         let a = [
-            1.0_f32, 2.0, 5.0, 6.0, 7.0, 3.0, 2.0, 1.0, -0.5, 5.0, 6.0, 7.0, 8.0, 9.0, 1.0, 2.0,
+            -2.0_f32, 2.0, 5.0, 6.0, 7.0, 3.0, 2.0, 1.0, -0.5, 5.0, 6.0, 7.0, 8.0, 9.0, 1.0, 2.0,
```

Comment:
Thanks for fixing this @heiher 

Could you add additional tests for this case, so that we don't lose test coverage?

Reply:
I've added new test cases to cover this issue. (BTW, in the previous patch, the changes to the f32x16 test cases didn't actually catch the problem. :joy: